src/io/unix/sync.rs

```diff
@@ -0,0 +1,197 @@
+// Sync linux
+
+use crate::prelude::*;
+use bytes::{Bytes, BytesMut};
+use nix::errno::Errno;
+use nix::fcntl::FcntlArg;
+use std::sync::atomic::AtomicU64;
+use std::{
+    os::unix::prelude::{AsRawFd, FileExt},
+    time::SystemTime,
+};
+use tokio::fs::OpenOptions;
+
+/// IO driver for file operations
+#[derive(Debug, Clone, Default)]
+pub struct IODriver;
+
+impl IODriver {
```

Comment:
Same as for async version: we need explicit constructors. Also it is good to remove `Default` implementation
1. `new_sync() -> Self`
2. `new() -> Self` - calls new_sync()